### PR TITLE
Enforce py3 wheels only in CI

### PR DESCRIPTION
**Pull request by scbedd**
@lmazuel  This is the "hard" break on py27.

It extends the `verifywhl` tox environment to do the additional following checks.

1. If no `python_requires` is present on your setup.py, it will error.
2. If a `python_requires` is defined, and is not `>= 3.0`, it will error.
3. If a `python_requires` is defined and set at something `>= 3.0`, but a `setup.cfg` is still present with a `universal=1` config key, it will error.

All of the above are skippable by environment variable `IGNORE_WHEEL_PYVERSION_CHECK`.

We can merge this PR whenever we want this to start hard erroring on py2 packages. Once that's clean, I can easily submit a PR to remove py2.7 from the build matrix.

### CODE REVIEW

**Comment by scbedd** [comment]
This PR will probably be green, but we should not merge immediately until we're ready for errrors to start popping up.

This PR will be green because `verifywhl` is not run as part of the PR suite. (that would be `whl, sdist, mindependency`.

File: eng/tox/verify_whl.py [hunk 1]
```diff
@@ -107,16 +106,33 @@ def should_verify_package(package_name):
...
+            exit(1)
+        else:
+            if not parse_version(trim_spec(python_requires)) >= parse_version("3.0"):
+                logging.info(
+                    "The python_requires value of '{}' should instead be at least '>=3.0'.".format(python_requires)
+                )
+                exit(1)
+            if "py2" in wheel_location:
+                logging.info(
+                    "The package {} is marked with 'python_requires{}', but a universal package was generated. Check your setup.cfg and ensure that 'universal=1' configuration is not present.".format(
```
Comment (lmazuel) [inline eng/tox/verify_whl.py:126]: `or remove the setup.cfg entirely`

File: eng/tox/verify_whl.py [hunk 2]
```diff
@@ -107,16 +106,33 @@ def should_verify_package(package_name):
...
-    pkg_name, namespace, ver = get_package_details(os.path.join(pkg_dir, "setup.py"))
+    pkg_name, namespace, ver, python_requires = get_package_details(os.path.join(pkg_dir, "setup.py"))
+
+    top_level_module = namespace.split(".")[0]
+    wheel_location = get_wheel(args.dist_dir, ver)
 
-    top_level_module = namespace.split('.')[0]
+    if not os.getenv("IGNORE_WHEEL_PYVERSION_CHECK"):
+        if not python_requires:
+            logging.info("Your package must define a setup argument for 'python_requires'.")
```
Comment (lmazuel) [inline eng/tox/verify_whl.py:116]: Can all errors message point to https://github.com/Azure/azure-sdk-for-python/blob/main/doc/dev/packaging.md additionally?

File: eng/tox/verify_whl.py [hunk 3]
```diff
@@ -107,16 +106,33 @@ def should_verify_package(package_name):
...
+            exit(1)
+        else:
+            if not parse_version(trim_spec(python_requires)) >= parse_version("3.0"):
+                logging.info(
+                    "The python_requires value of '{}' should instead be at least '>=3.0'.".format(python_requires)
```
Comment (lmazuel) [inline eng/tox/verify_whl.py:121]: Could we make it to at least 3.6?

Comment (scbedd) [reply eng/tox/verify_whl.py:121]: No problem! I was expecting points of clarity like this 👍

Comment (lmazuel) [reply eng/tox/verify_whl.py:121]: Can you store the `"3.0"` into a global variable, this way when we do 3.6, 3.7, etc. it's one update in one place? Something like
```python
MIN_PY_VERSION="3.6"

if not parse_version(trim_spec(python_requires)) >= parse_version(MIN_PY_VERSION):
```

**Review by lmazuel — APPROVED** [review]

**Comment by scbedd** [comment]
Just a reminder. This PR should only be merged when we are ready to flip `py3only` enforcement on. This PR is currently red due to a failure to verify the wheel of various packages in `core/`. 

Once those are fixed, a requeued check should come back green.

**Pull request closed without merging**